types/src/unit_tests/trusted_state_test.rs

```diff
@@ -197,7 +197,10 @@ fn arb_update_proof(
                         epoch += 1;
                         version += version_delta as u64;
 
-                        LedgerInfoWithSignatures::new(ledger_info, signatures)
+                        LedgerInfoWithSignatures::Ed25519(LedgerInfoWithEd25519::new(
```

Comment:
I think you can just impl LedgerInfoWithSignatures::new to be a wrapper around ledgerInfoWithEd25519